src/mono/mono/mini/mini-generic-sharing.c

```diff
@@ -4302,9 +4307,32 @@ mini_get_rgctx_entry_slot (MonoJumpInfoRgctxEntry *entry)
 	MonoJitMemoryManager *jit_mm = get_default_jit_mm ();
 
 	if (entry->in_mrgctx)
-		return lookup_or_register_info (jit_mm->mem_manager, entry->d.method->klass, entry->d.method, entry->in_mrgctx, entry_data, entry->info_type, mono_method_get_context (entry->d.method));
+		return lookup_or_register_info (jit_mm->mem_manager, entry->d.method->klass, entry->d.method, entry->in_mrgctx, entry_data, &did_register, entry->info_type, mono_method_get_context (entry->d.method));
 	else
-		return lookup_or_register_info (jit_mm->mem_manager, entry->d.klass, NULL, entry->in_mrgctx, entry_data, entry->info_type, mono_class_get_context (entry->d.klass));
+		return lookup_or_register_info (jit_mm->mem_manager, entry->d.klass, NULL, entry->in_mrgctx, entry_data, &did_register, entry->info_type, mono_class_get_context (entry->d.klass));
+
+	if (!did_register)
+		switch (entry->data->type) {
+		case MONO_PATCH_INFO_GSHAREDVT_CALL:
+			g_free ((MonoJumpInfoGSharedVtCall *) entry_data);
+			break;
+		case MONO_PATCH_INFO_VIRT_METHOD:
+			g_free ((MonoJumpInfoVirtMethod *) entry_data);
+			break;
+		case MONO_PATCH_INFO_DELEGATE_TRAMPOLINE:
+			g_free ((MonoDelegateClassMethodPair *) entry_data);
+			break;
+		case MONO_PATCH_INFO_GSHAREDVT_METHOD: {
+			for (int i = 0; i < ((MonoGSharedVtMethodInfo *) entry_data)->num_entries; ++i) {
+				g_free (&((MonoGSharedVtMethodInfo *) entry_data)->entries[i]);
+			}
```

Comment:
```suggestion
			g_free (((MonoGSharedVtMethodInfo *) entry_data)->entries);
```

It's one malloc - so there should be one free.